app/services/retrieve_pipeline_viz_graph_data_service.rb

```diff
@@ -64,26 +55,66 @@ def create_stage_nodes_scaffolding
     stages = @all_dag_jsons.map.with_index do |dag_json, stage_index|
       stage_step_statuses = all_step_statuses[stage_index]
       stage_step_descriptions = STEP_DESCRIPTIONS[@stage_names[stage_index]]["steps"]
+
+      all_redefined_statuses = []
       steps = dag_json["steps"].map do |step|
         status_info = stage_step_statuses[step["out"]] || {}
         description = status_info["description"].blank? ? stage_step_descriptions[step["out"]] : status_info["description"]
+        status = redefine_job_status(status_info["status"])
+        all_redefined_statuses << status
         {
           name: modify_step_name(step["out"]),
           description: description,
           inputEdges: [],
           outputEdges: [],
-          status: JOB_STATUS_NUM_TO_STRING[status_info["status"]],
+          status: status,
+          startTime: status_info["start_time"],
         }
       end
 
       {
         steps: steps,
-        jobStatus: dag_json[:job_status],
+        jobStatus: stage_job_status(all_redefined_statuses),
       }
     end
     return stages
   end
 
+  def step_statuses
+    @pipeline_run.pipeline_run_stages.map do |prs|
+      begin
+        JSON.parse(get_s3_file(prs.step_status_file_path) || "{}")
+      rescue JSON::ParserError
+        {}
+      end
+    end
+  end
+
+  def redefine_job_status(status)
+    case status
+    when "instantiated", nil
+      "notStarted"
+    when "running", "finished_running"
```

Comment:
finished_running = inProgress?

Reply:
Just added a comment, but `finished_running` occurs when the outputs have been created, but hasn't been uploaded to aws yet. Since the file is not available to download yet, I marked as "inProgress"